// react-native-turbo-mock-location-detector@2.3.1/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isMockingLocation = exports.MockLocationDetectorErrorCode = void 0;
var _reactNative = require("react-native");
const LINKING_ERROR = 'The package \'react-native-mock-location-detector\' doesn\'t seem to be linked. Make sure: \n\n' + _reactNative.Platform.select({
  ios: '- You have run pod install\'\n',
  default: ''
}) + '- You rebuilt the app after installing the package\n' + '- You are not using Expo managed workflow\n';

// @ts-expect-error
const isTurboModuleEnabled = global.__turboModuleProxy != null;
const MockLocationDetectorModule = isTurboModuleEnabled ? require('./NativeMockLocationDetector').default : _reactNative.NativeModules.MockLocationDetector;
const MockLocationDetector = MockLocationDetectorModule ? MockLocationDetectorModule : new Proxy({}, {
  get() {
    throw new Error(LINKING_ERROR);
  }
});
let MockLocationDetectorErrorCode = /*#__PURE__*/function (MockLocationDetectorErrorCode) {
  MockLocationDetectorErrorCode["GPSNotEnabled"] = "0";
  MockLocationDetectorErrorCode["NoLocationPermissionEnabled"] = "1";
  MockLocationDetectorErrorCode["CantDetermine"] = "2";
  return MockLocationDetectorErrorCode;
}({});
exports.MockLocationDetectorErrorCode = MockLocationDetectorErrorCode;
const isMockingLocation = () => {
  return MockLocationDetector.isMockingLocation().catch(_ref => {
    let {
      code,
      message
    } = _ref;
    throw {
      code,
      message
    };
  });
};
exports.isMockingLocation = isMockingLocation;
