// react-native-turbo-mock-location-detector@2.3.1/lib/module/index.js
import { NativeModules, Platform } from 'react-native';
const LINKING_ERROR = 'The package \'react-native-mock-location-detector\' doesn\'t seem to be linked. Make sure: \n\n' + Platform.select({
  ios: '- You have run pod install\'\n',
  default: ''
}) + '- You rebuilt the app after installing the package\n' + '- You are not using Expo managed workflow\n';

// @ts-expect-error
const isTurboModuleEnabled = global.__turboModuleProxy != null;
const MockLocationDetectorModule = isTurboModuleEnabled ? require('./NativeMockLocationDetector').default : NativeModules.MockLocationDetector;
const MockLocationDetector = MockLocationDetectorModule ? MockLocationDetectorModule : new Proxy({}, {
  get() {
    throw new Error(LINKING_ERROR);
  }
});
export let MockLocationDetectorErrorCode = /*#__PURE__*/function (MockLocationDetectorErrorCode) {
  MockLocationDetectorErrorCode["GPSNotEnabled"] = "0";
  MockLocationDetectorErrorCode["NoLocationPermissionEnabled"] = "1";
  MockLocationDetectorErrorCode["CantDetermine"] = "2";
  return MockLocationDetectorErrorCode;
}({});
export const isMockingLocation = () => {
  return MockLocationDetector.isMockingLocation().catch(_ref => {
    let {
      code,
      message
    } = _ref;
    throw {
      code,
      message
    };
  });
};
